### DISCUSSION TITLE
Writing seperate configs for optimizer

### DISCUSSION THREAD

**Discussion by hyenal** [martinsbruveris/tensorflow-image-models#38] [Ideas]
Right now we are defining optimizers with quite a number of wasteful parameters. However in the line below we know that for each optimizer we **MUST** request:
- the optimizer class (see line cited)
- the schedule class

But this is exactly the use case within scope of `cfg_serializable`

Then the idea would be to create config schedule and solver classes so that we register only the parameters for this class. 

We should however document it in the library so that it is more user friendly but that could be a nice way to get rid of unused parameters and confusing name we had so far I feel :) 

https://github.com/martinsbruveris/tensorflow-image-models/blob/b47f2a042869698284626154d98ed91930e16e41/tfimm/train/optimizer.py#L13

**Comment by martinsbruveris** [comment]
Yes, I see your point. It is definitely the case right now that different LR schedules use different parameters and so we do end up with unused parameters for all the schedules we are not using... For optimisers this problem is less relevant, because they mostly end up using the same parameters. For LR schedules, your idea is probably the way to go.
